portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-validator.js

```diff
@@ -82,8 +82,45 @@ const validateCsvData = (csvData) => {
   return validationErrors;
 };
 
+const filenameContainsReportingPeriod = (filename) => {
+  let containsReportPeriod = false;
+  let monthInFilename;
+  // Do I want to be case sensitive?
+  Object.values(MONTH_NAMES).some(({ longName, shortName }) => {
+    containsReportPeriod = filename.includes(longName) || filename.includes(shortName);
+    if (containsReportPeriod) {
+      monthInFilename = longName;
+    }
+    return containsReportPeriod;
+  });
+  return { containsReportPeriod, monthInFilename };
+};
+
+const validateFilenameContainsReportPeriod = (filename, currentReportPeriod) => {
+  const expectedFilenameReportPeriod = currentReportPeriod.replace(' ', '_');
+
+  const { containsReportPeriod, monthInFilename } = filenameContainsReportingPeriod(filename);
+  if (!containsReportPeriod) {
+    const filenameError = `The selected file must contain the reporting period as part of its name, for example '${expectedFilenameReportPeriod}'`;
+    return { filenameError };
+  }
+
+  const currentReportPeriodYear = currentReportPeriod.split(' ').at(-1); // Expected format 'MMMM yyyy'
+  const filenameReportPeriod = `${monthInFilename}_${currentReportPeriodYear}`;
+  const filenameReportPeriodMatches = filenameReportPeriod === expectedFilenameReportPeriod;
```

Comment:
I don't think this validates that the filename contains the correct year, as we're just using the expected year in `filenameReportPeriod`?

If we can ignore the need to check for month short names I think this logic could be a lot simpler. e.g.
```
const validateFilenameContainsReportPeriod = (filename, currentReportPeriod) => {
  const expectedFilenameReportPeriod = currentReportPeriod.replace(' ', '_');

  const filenameContainsReportPeriod = filename.toLowerCase().includes(expectedFilenameReportPeriod.toLowerCase());

  if (!filenameContainsReportPeriod) {
    const filenameError = `The selected file must contain the current reporting period as part of its name - i.e. '${expectedFilenameReportPeriod}'`;
    return { filenameError };
  }

  return {};
};
```
If we do need to also check the short name we should be able to extract the long name from the `currentReportPeriod`, map to the short name, and check that too.

Reply:
I have refactored this quite significantly so it actually checks the format we expect now - let me know what you